src/main/java/pl/javastart/restoffers/offers/Offer.java

```diff
@@ -0,0 +1,73 @@
+package pl.javastart.restoffers.offers;
+
+import jakarta.persistence.*;
+import pl.javastart.restoffers.categories.Category;
+
+@Entity
+public class Offer {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String title;
+
+    private String description;
+
+    private String imgUrl;
+
+    private double price;
+
+    private String categoryName;
```

Comment:
Na poziomie bazy danych tutaj powinien być klucz obcy, czyli id kategorii, albo tak, jak masz poniżej komentarz, czyli relacja zwrotna. W takiej sytuacji niezbędne jest oczywiście dodanie klasy DTO, aby dostosować odpowiedzi z punktów krańcowych

Reply:
Tak, próbowałem wykonać to w ten sposób. Miałem nawet klasę DTO dla oferty jednak nie potrafiłem, powiązać tej klasy w momencie kiedy do tworzenia obiektu w bazie danych potrzebny był obiekt typu Category. Kombinowałem coś, żeby po Stringu konwertować wprowadzoną kategorię na obiekt Category, niestety nie udało mi się to. Pewnie jakbym jeszcze pokombinował to coś by się urodziło, jednak z uwagi na ograniczoną ilość czasu postanowiłem zrezygnować z klasy DTO. Po prostu sobie ułatwiłem. Zdaję sobie sprawę, że w komercyjnym kodzie by to nie przeszło :)